app/models/course/assessment/answer.rb

```diff
@@ -17,6 +17,7 @@ class Course::Assessment::Answer < ActiveRecord::Base
   validate :validate_assessment_state, if: :attempting?
   validates :grade, presence: true, unless: :attempting?
   validates :grade, absence: true, if: :attempting?
+  validate :validate_consistent_grade, unless: :attempting?
```

Comment:
`validates :grade, allow_nil: true, numericality: { less_than_or_equal_to: :maximum_grade }` Does this work ?